Src/Test.UnitTests.Sarif.PatternMatcher.Cli/AnalyzeCommandTests.cs

```diff
@@ -277,6 +280,98 @@ public void AnalyzeCommand_ShouldThrowExceptionForConflictingFileSizeOptions()
+                    expectedNoValidation = FailureLevel.Warning,
+                    expectedStaticOnly = FailureLevel.Warning,
+                    expectedStaticDynamic = FailureLevel.Warning
+                },
+                new {
+                    jsonLevel = FailureLevel.Error,
+                    dynamicValidationEnabled = false,
+                    expectedNoValidation = FailureLevel.Error,
+                    expectedStaticOnly = FailureLevel.Error,
+                    expectedStaticDynamic = FailureLevel.Warning
+                },
+                new {
+                    jsonLevel = FailureLevel.Note,
+                    dynamicValidationEnabled = true,
+                    expectedNoValidation = FailureLevel.Note,
+                    expectedStaticOnly = FailureLevel.Note,
+                    expectedStaticDynamic = FailureLevel.Error
+                },
+                new {
+                    jsonLevel = FailureLevel.Warning,
+                    dynamicValidationEnabled = true,
+                    expectedNoValidation = FailureLevel.Warning,
+                    expectedStaticOnly = FailureLevel.Warning,
+                    expectedStaticDynamic = FailureLevel.Error
+                },
+                new {
+                    jsonLevel = FailureLevel.Error,
+                    dynamicValidationEnabled = true,
+                    expectedNoValidation = FailureLevel.Error,
+                    expectedStaticOnly = FailureLevel.Error,
+                    expectedStaticDynamic = FailureLevel.Error
+                }
+            };
+
+            foreach (var testCase in testCases)
+            {
+                string definitionsText = GetSingleLineRuleDefinitionFailureLevel(testCase.jsonLevel);
+                string testScenarioName = "NoValidation";
+                SarifLog logFile = RunAnalyzeCommandWithDynamicValidation(definitionsText,
+                                                                          testScenarioName,
+                                                                          testCase.dynamicValidationEnabled);
+
+                ValidateFailureLevelByValidationType(expectedFailureLevel: testCase.expectedNoValidation,
+                                                     sarifLog: logFile,
+                                                     validationScenario: testScenarioName,
+                                                     isDynamicAnalysis: testCase.dynamicValidationEnabled,
+                                                     ref sb);
+
+                testScenarioName = "StaticOnly";
+                logFile = RunAnalyzeCommandWithDynamicValidation(definitionsText,
+                                                                 testScenarioName,
+                                                                 testCase.dynamicValidationEnabled);
+
+                ValidateFailureLevelByValidationType(expectedFailureLevel: testCase.expectedStaticOnly,
```

Comment:
>testCase.expectedStaticOnly [](http://example.com/codeflow?start=75&length=27)

This was the test gap, we were validating against the previous test case instead of the current test case.